### ISSUE TITLE
`build-docker` GH Actions workflow fails on master

### ISSUE THREAD

**Issue by tk-woven** [TRI-ML/dgp#106]
# Description

The `build-docker` workflow fails on master with

```
#4 [internal] load metadata for docker.io/nvidia/cuda:11.1-devel-ubuntu18.04
#4 ERROR: docker.io/nvidia/cuda:11.1-devel-ubuntu18.04: not found
------
 > [internal] load metadata for docker.io/nvidia/cuda:11.1-devel-ubuntu18.04:
------
error: failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to create LLB definition: docker.io/nvidia/cuda:11.1-devel-ubuntu18.04: not found
Error: buildx failed with: error: failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to create LLB definition: docker.io/nvidia/cuda:11.1-devel-ubuntu18.04: not found
```

# Reproduction

Run the `build-docker` workflow on master manually or trigger it via a merge to master.

# Fix

Update the base image in our Dockerfile.

**Formerly:**

```
FROM nvidia/cuda:11.1-devel-ubuntu18.04
```

I don't see this entry in the [docker registry](https://hub.docker.com/r/nvidia/cuda/tags?page=1&name=11.1-devel). I suspect it was renamed or replaced with alternatives.

**Fix:**

```
FROM nvidia/cuda:11.1.1-devel-ubuntu18.04
```

The new image is [here](https://hub.docker.com/layers/cuda/nvidia/cuda/11.1.1-devel-ubuntu18.04/images/sha256-3cb854e8d05177d7cb1ede24180ae3e12c9b0e307d6af0fabd3579fe978698e5?context=explore).

**Comment by tk-woven** [comment]
For those who can see it, [here](https://github.com/TRI-ML/dgp/runs/7645076177) is an example of a failed run.

**Comment by tk-woven** [comment]
Fixed in #107 .
